tasks/cluster.py

```diff
@@ -56,12 +62,16 @@ def provision(ctx):
     _run_aks_cmd(
         "create",
         [
-            "--name {}".format(AKS_CLUSTER_NAME),
-            "--node-count {}".format(AKS_CLUSTER_NODE_COUNT),
-            "--node-vm-size {}".format(AZURE_K8S_VM_SIZE),
+            "--name {}".format(AZURE_K8S_CLUSTER_NAME),
+            "--node-count {}".format(node_count),
+            "--node-vm-size {}".format(vm_size),
             "--os-sku Ubuntu",
             "--kubernetes-version {}".format(k8s_ver),
             "--ssh-key-value {}".format(AZURE_PUB_SSH_KEY),
+            "--location {}".format(location),
+            "{}".format(
+                "--enable-addons confcom" if "Standard_DC" in vm else ""
+            ),
```

Comment:
What does this part mean? Is this something to do with SGX? I can see that if `Standard_DC` implies SGX then an `sgx` parameter would be unnecessary, but without a comment this is a little unclear. I propose the following:

- Add a `--sgx` flag to the `provision` task.
- Check that, if `sgx` is enabled, `Standard_DC` is in the `vm` variable, and error if not. The error text would be something like `At the time of writing only Standard_DC machines support SGX`.

It makes the SGX provisioning a tiny bit more verbose as you have to add the `--sgx` flag but I think it's more flexible (in case more than just `Standard_DC` machines support SGX), and clearer to the reader.